addons/crm/models/res_partner.py

```diff
@@ -60,9 +64,5 @@ def action_view_opportunity(self):
         '''
         action = self.env['ir.actions.act_window']._for_xml_id('crm.crm_lead_opportunities')
         action['context'] = {}
-        if self.is_company:
-            action['domain'] = [('partner_id.commercial_partner_id', '=', self.id)]
-        else:
-            action['domain'] = [('partner_id', '=', self.id)]
-        action['domain'] = expression.AND([action['domain'], [('active', 'in', [True, False])]])
+        action['domain'] = expression.AND([self._get_opportunity_domian(), [('active', 'in', [True, False])]])
```

Comment:
Is it really equivalent ? Needs a bit more :coffee: but unsure it is always the same ?

Reply:
following my last log-note and from what I understand, 
when a partner click on the stat button whether the partner is a company or individual all of his and it's child opportunities should be shown but due to the ` if self.is_company` it only works when the partner is company but if hierarchy is created and the parent is individual then the current code is breaking so we need a domain which will search all the opportunities where either partner is set as customer or assign to it or it's child is set as customer or assign to it